File: CS/EntityFrameworkCodeFirstBindingExample/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Scheduling;
using System.Collections.ObjectModel;
using System.Data.Entity;

namespace EntityFrameworkCodeFirstBindingExample
{
    [POCOViewModel]
    public class MainViewModel {

        #region #instances
        MySchedulerModel mySchedulerContext;
        public virtual ObservableCollection<Car> Resources { get; set; }
        public virtual ObservableCollection<CarScheduling> Appointments { get; set; }
        #endregion #instances

        #region #savechanges
        public void AppointmentsUpdated()
        {
            mySchedulerContext.SaveChanges();
        }
        #endregion #savechanges

        #region #InitNewAppointment
        public void InitNewAppointment(AppointmentItemEventArgs args)
        {
            args.Appointment.Reminders.Clear();
        }
        #endregion #InitNewAppointment

        #region #filldata
        protected MainViewModel() {
            mySchedulerContext = new MySchedulerModel();
            mySchedulerContext.Cars.Load();
            Resources = mySchedulerContext.Cars.Local;
            mySchedulerContext.CarSchedulings.Load();
            Appointments = mySchedulerContext.CarSchedulings.Local;
        }
        #endregion #filldata
        public static MainViewModel Create() {
            return ViewModelSource.Create(() => new MainViewModel());
        }
    }
}

File: VB/EntityFrameworkCodeFirstBindingExample/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Scheduling
Imports System.Collections.ObjectModel
Imports System.Data.Entity

Namespace EntityFrameworkCodeFirstBindingExample
    <POCOViewModel> _
    Public Class MainViewModel

        #Region "#instances"
        Private mySchedulerContext As MySchedulerModel
        Public Overridable Property Resources() As ObservableCollection(Of Car)
        Public Overridable Property Appointments() As ObservableCollection(Of CarScheduling)
        #End Region ' #instances

        #Region "#savechanges"
        Public Sub AppointmentsUpdated()
            mySchedulerContext.SaveChanges()
        End Sub
        #End Region ' #savechanges

        #Region "#InitNewAppointment"
        Public Sub InitNewAppointment(ByVal args As AppointmentItemEventArgs)
            args.Appointment.Reminders.Clear()
        End Sub
        #End Region ' #InitNewAppointment

        #Region "#filldata"
        Protected Sub New()
            mySchedulerContext = New MySchedulerModel()
            mySchedulerContext.Cars.Load()
            Resources = mySchedulerContext.Cars.Local
            mySchedulerContext.CarSchedulings.Load()
            Appointments = mySchedulerContext.CarSchedulings.Local
        End Sub
        #End Region ' #filldata
        Public Shared Function Create() As MainViewModel
            Return ViewModelSource.Create(Function() New MainViewModel())
        End Function
    End Class
End Namespace
